File: src/tg37/tinyc/eq01.c
/* SITUATION: Parsed an assignment expression. Two stack entries, lvalue, datam.
 *	Effects the assignment. 
 */
void _eq() {
	int  iDatum;  /* memcpy into these from pr using val.stuff */
	char cDatum;  /*  and val.size, giving needed cast */
	void* pDatum;
	void* where;

	struct stackentry *val = &stack[nxtstack-1]; /* value (on top) */
	struct stackentry *lval = &stack[nxtstack-2]; /* where to put it */
	if(verbose[VE]){
		fprintf(stderr,"\neq: lval");
		dumpStackEntry(nxtstack-2);
		fprintf(stderr,"\neq: val");
		dumpStackEntry(nxtstack-1);
	}
	popst();popst();  
	where = &((*lval).value.up);
	int class = (*lval).class;
	int type = (*lval).type;
//	int whereSize = typeToSize(class,type);  /* of the lvalue */

	if((*lval).lvalue != 'L') { 
		eset(LVALERR); 
		return; 
	}
	
	else if(class==0 && (*val).class==0) {
		if(type==Int){
			if( (*val).lvalue=='L' ) {
				iDatum = get_int((*val).value.up);
			}
			else {
				iDatum = (*val).value.ui;
			}
			if((*val).type==Char) iDatum = iDatum&0xff;
			put_int( (*lval).value.up, iDatum);
			pushk(iDatum);
		}
	}
}

File: src/tg37/tinyc/Eq01.java
/*	Tests for and parses constant. Cursor moved just beyond constant.
 */
package tg37.tinyc;

public class Eq01 extends PT {
	static Stack stk;
	
    private static Eq01 instance;
    private Eq01(){}
    public static synchronized Eq01 getInstance(){
        if(instance == null){
            instance = new Eq01();
			stk = Stack.getInstance();
//			stmt = ST.getInstance();
//			vt = Vartab.getInstance();
//			dl = Dialog.getInstance();
        }
        return instance;
    }

/* SITUATION: Parsed an assignment expression. Two stack entries, lvalue, datum.
 *	Effects the assignment. 
 */
	public static void eq() {
		if(TJ.traceON);
//stk.dump("\nEq01~26, stk before pops: ");
		int value = stk.toptoi();
		Stuff lval = stk.popst();
		lval.setInt(value);
		stk.pushStuff(lval);
		if(TJ.traceON)stk.dump("Eq01~32, After assignment push: \n");
//stk.dump("\nEq01~32, After assignment push: ");
//System.err.println("");
	}
}
